src/app/app-services/courses.service.ts

```diff
@@ -12,8 +14,9 @@ export class CoursesService {
     @Inject('BASE_URL') private readonly BASE_URL: string
   ) { }
 
-  list(params: { start?: string, count?: string, textFragment?: string }): Observable<Course[]> {
-    return this.httpClient.get<Course[]>(`${this.BASE_URL}/courses`, { params });
+  list(params?: CoursesServiceListParams): Observable<Course[]> {
+    return this.httpClient.get<Course[]>(`${this.BASE_URL}/courses`, { params: params as CoursesServiceIndexedParams })
```

Comment:
We could have left this line unchanged couldn't we?

Reply:
Yes, that's true.

The reason behind this change is that I tried to make this service and its related Action-Effect (listCourses & listCourses$) future-proof, I mean, if for some reason we have to change any of the allowed params, this interface (CoursesServiceListParams) will ensure that we have changed both here and in the action.

This could be not the best approach though.

Reply:
The approach depends...

Here it's ok, but, eventually, we may not have a good naming coming from the API, so we could have interfaces/classes to be used to in request/response from API and then have another set of interfaces/classes to be used elsewhere in application (with the naming and proper data structures).

In one of my previous projects, we used classes instead of interfaces. We then caught the response object and passed as argument of the class constructor. In the constructor we had some transformation logic, like, parsing strings to date properties etc.

Reply:
Though the reason for my comment was not line 17, but, line 18. We could keep it as is without the need for `params as CoursesServiceListParams`. Did you try it?